### PR TITLE
Re-introduce workaround to fix ADO path names for PR eval

### PR DESCRIPTION
**Pull request by cfernald**
Adds a work-around to the naming provided by System.PullRequest.targetBranch on ADO repos by stripping the refs/heads/ from the path since that makes it unrecognizable to git for the git diff used by stuart_pr_eval.

### CODE REVIEW

File: Steps/PrGate.yml [hunk 1]
```diff
@@ -87,12 +87,17 @@ steps:
 
 # Trim the package list if this is a PR
 - ${{ if eq(parameters.do_pr_eval, true) }}:
+  - powershell:
```
Comment (kuqin12) [inline Steps/PrGate.yml:90]: nit: :)

**Review by kuqin12 — APPROVED** [review]

**Comment by cfernald** [comment]
I have been defeated by azure pipeline expressions. I attempted to do this inline in existing step, but could not get the combination of the replace expression with the pre-defined system variable to function properly. If anyone has any ideas on this please let me know.

**Review by os-d — APPROVED** [review]

File: Steps/PrGate.yml (hunk 1 above)
Comment (cfernald) [reply Steps/PrGate.yml:90]: Sorry, not following what the nit is about here

**Review by makubacki — APPROVED** [review]

**Comment by makubacki** [comment]
@cfernald what about the isntance in https://github.com/microsoft/mu_devops/blob/main/Steps/BuildPlatform.yml#L84?

**Comment by cfernald** [comment]
> @cfernald what about the instance in https://github.com/microsoft/mu_devops/blob/main/Steps/BuildPlatform.yml#L84?

good point, will fix there as well

**Pull request merged**